src/Build/Logging/FancyLogger/FancyLogger.cs

```diff
@@ -0,0 +1,161 @@
+            // If id already exists...
+            if (projectConsoleLines.ContainsKey(id)) return;
+            // Create line
+            FancyLoggerBufferLine line = FancyLoggerBuffer.WriteNewLine(
+                ANSIBuilder.Alignment.SpaceBetween(
+                    $"{ANSIBuilder.Graphics.Spinner()} {ANSIBuilder.Formatting.Dim("Project - ")} {GetUnambiguousPath(e.ProjectFile!)}",
+                    "(5 targets completed)",
+                    Console.WindowWidth
+                )
+            );
+
+            projectConsoleLines.Add(id, line);
+            Thread.Sleep(400);
+        }
+        void eventSource_ProjectFinished(object sender, ProjectFinishedEventArgs e)
+        {
+            // Get project id
+            int id = e.BuildEventContext!.ProjectInstanceId;
+            // If id does not exist
+            if (!projectConsoleLines.ContainsKey(id)) return;
+            // Get line id
+            FancyLoggerBufferLine line = projectConsoleLines[id];
+            // Update line
+            FancyLoggerBuffer.UpdateLine(line.Id,
+                ANSIBuilder.Alignment.SpaceBetween(
+                    $"{ANSIBuilder.Formatting.Color("✓", ANSIBuilder.Formatting.ForegroundColor.Green)} {ANSIBuilder.Formatting.Dim("Project - ")} {ANSIBuilder.Formatting.Color(GetUnambiguousPath(e.ProjectFile!), ANSIBuilder.Formatting.ForegroundColor.Green)}",
+                    "(5 targets completed)",
+                    Console.WindowWidth
+                )
+            );
+            /*// TODO
+            int id = e.BuildEventContext!.ProjectInstanceId;
+            var line = projectConsoleLines[id];
+            FancyLoggerBuffer.UpdateLine(line.Id, ""
+                + "D"
+                + " "
+                + ANSIBuilder.Formatting.Dim("Project - ")
+                + GetUnambiguousPath(e.ProjectFile)
+            );*/
+        }
+        // Target
+        void eventSource_TargetStarted(object sender, TargetStartedEventArgs e)
+        {
+        }
+        void eventSource_TargetFinished(object sender, TargetFinishedEventArgs e)
+        {
+        }
+
+        // Task
+        void eventSource_TaskStarted(object sender, TaskStartedEventArgs e)
+        {
+            existingTasks++;
+        }
+
+        void eventSource_TaskFinished(object sender, TaskFinishedEventArgs e)
+        {
+            completedTasks++;
+        }
+
+        void eventSource_MessageRaised(object sender, BuildMessageEventArgs e)
+        {
+        }
+        void eventSource_WarningRaised(object sender, BuildWarningEventArgs e)
+        {
+            // Console.WriteLine("Warning raised");
+            FancyLoggerBuffer.WriteNewLine("Warning!");
+        }
+        void eventSource_ErrorRaised(object sender, BuildErrorEventArgs e)
+        {
+            // TODO: Try to redirect to stderr
+            // Console.WriteLine("Error raised");
+            FancyLoggerBuffer.WriteNewLine("Error");
```

Comment:
Don't take this suggestion:

```suggestion
            FancyLoggerBuffer.WriteNewLine("Error 😱");
```